File: js/packages/app/src/__tests__/features/classes/polywrap/classes-example/src/polywrap/external/module/WrapModule.ts
import { IExternalWrapInstance } from "@polywrap/reim-wrap-js";
import { ImportBindings } from "./ImportBindings";


import { create as createTestClass } from "../classes/TestClass";


export class WrapModule {
  static wrapInstance: IExternalWrapInstance | null;

  static connect(instance: IExternalWrapInstance): void {
    WrapModule.wrapInstance = instance;
  }

  static import(instance: IExternalWrapInstance): ImportBindings {
    return new ImportBindings(

                                          
      createTestClass(instance),
      
    );
  }
}

File: rust/test_cases/wrappers/external_references/src/polywrap/external/module/wrap_module.rs
use std::sync::Arc;

use reim_dt::ExternalModule;
use crate::polywrap::external::module::import_bindings::{ ImportBindings };

use crate::polywrap::external::global_functions::test_external_global_function::{ create as createTestExternalGlobalFunction };

use crate::polywrap::external::classes::test_external_class::{ create as createTestExternalClass };

use crate::polywrap::external::module::external_wrap_module::set_external_module;

pub struct WrapModule {}

impl WrapModule {
  pub async fn connect(external_module: Arc<dyn ExternalModule>) {
    set_external_module(external_module).await;
  }

  pub fn import(instance: Arc<dyn ExternalModule>) -> ImportBindings {
    ImportBindings::new(
                                                      
      createTestExternalGlobalFunction(Arc::clone(&instance)),
      
                  
      createTestExternalClass(Arc::clone(&instance)),
      
    )
  }
}
